flanker/addresslib/address.py

```diff
@@ -311,8 +313,17 @@ def validate_address(addr_spec, metrics=False):
     # lookup custom local-part grammar if it exists
     bstart = time()
     plugin = plugin_for_esp(exchanger)
+
+    # Since yahoo has managed mailbox rules differ from hosted we have to
```

Comment:
This is Yahoo specific logic therefore it should be moved to the yahoo plugin. The yahoo plugin should check if the address is "managed" and apply respective patterns. This module should not be modified.